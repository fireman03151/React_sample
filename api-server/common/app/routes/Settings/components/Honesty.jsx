import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button, Panel } from 'react-bootstrap';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';

import { FullWidthRow } from '../../../helperComponents';
import SectionHeader from './SectionHeader.jsx';
import { userSelector } from '../../../redux';
import academicPolicy from '../../../../resource/academicPolicy';
import { updateUserBackend } from '../redux';

const propTypes = {
  isHonest: PropTypes.bool,
  policy: PropTypes.arrayOf(PropTypes.string),
  updateUserBackend: PropTypes.func.isRequired
};

const mapStateToProps = createSelector(
  userSelector,
  ({ isHonest }) => ({
    policy: academicPolicy,
    isHonest
  })
);

const mapDispatchToProps = { updateUserBackend };

class Honesty extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      showHonesty: false
    };

    this.handleAgreeClick = this.handleAgreeClick.bind(this);
  }

  handleAgreeClick() {
    this.props.updateUserBackend({ isHonest: true });
  }

  render() {
    const { policy, isHonest } = this.props;
    const isHonestAgreed = (
      <Panel bsStyle='info'>
        <p>
          You have accepted our Academic Honesty Policy.
        </p>
      </Panel>
    );
    const agreeButton = (
      <Button
        block={ true }
        bsStyle='primary'
        onClick={ this.handleAgreeClick }
        >
        Agree
      </Button>
    );
    return (
      <div className='honesty-policy'>
        <SectionHeader>
          Academic Honesty Policy
        </SectionHeader>
        <FullWidthRow>
          <Panel>
            {
              policy.map(
                (line, i) => (
                  <p
                  dangerouslySetInnerHTML={{ __html: line }}
                  key={ '' + i + line.slice(0, 10) }
                  />
                )
              )
            }
            <br />
            {
              isHonest ?
                isHonestAgreed :
                agreeButton
            }
          </Panel>
        </FullWidthRow>
      </div>
    );
  }
}


Honesty.displayName = 'Honesty';
Honesty.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(Honesty);
