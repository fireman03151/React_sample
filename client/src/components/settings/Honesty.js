import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Panel } from '@freecodecamp/react-bootstrap';

import { FullWidthRow } from '../helpers';
import SectionHeader from './SectionHeader';
import HonestyPolicy from '../../resources/honesty-policy';

import './honesty.css';

const propTypes = {
  isHonest: PropTypes.bool,
  policy: PropTypes.arrayOf(PropTypes.string),
  updateIsHonest: PropTypes.func.isRequired
};

class Honesty extends Component {
  handleAgreeClick = () => this.props.updateIsHonest({ isHonest: true });

  renderAgreeButton = () => (
    <Button block={true} bsStyle='primary' onClick={this.handleAgreeClick}>
      Agree
    </Button>
  );

  renderIsHonestAgreed = () => (
    <Panel bsStyle='info' className='agreed btn'>
      <p>You have accepted our Academic Honesty Policy.</p>
    </Panel>
  );

  render() {
    const { isHonest } = this.props;

    return (
      <section className='honesty-policy'>
        <SectionHeader>Academic Honesty Policy</SectionHeader>
        <FullWidthRow>
          <Panel className='honesty-panel'>
            <HonestyPolicy />
          </Panel>
          <br />
          {isHonest ? this.renderIsHonestAgreed() : this.renderAgreeButton()}
        </FullWidthRow>
      </section>
    );
  }
}

Honesty.displayName = 'Honesty';
Honesty.propTypes = propTypes;

export default Honesty;
