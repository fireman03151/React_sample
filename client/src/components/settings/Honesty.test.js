/* global expect jest */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';

import Honesty from './Honesty';
import { Button } from '@freecodecamp/react-bootstrap';

describe('<Honesty />', () => {
  const renderer = new ShallowRenderer();
  const updateIsHonestMock = jest.fn();

  test('<Honesty /> snapshot when isHonest is false', () => {
    const componentToRender = (
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    );
    const component = renderer.render(componentToRender);
    expect(component).toMatchSnapshot('Honesty');
  });

  test('<Honesty /> snapshot when isHonest is true', () => {
    const componentToRender = (
      <Honesty isHonest={true} updateIsHonest={updateIsHonestMock} />
    );
    const component = renderer.render(componentToRender);
    expect(component).toMatchSnapshot('HonestyAccepted');
  });

  test('should call updateIsHonest method on clicking agree button', () => {
    const root = TestRenderer.create(
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    ).root;

    root.findByType(Button).props.onClick();
    expect(updateIsHonestMock).toHaveBeenCalledWith({ isHonest: true });
  });
});
