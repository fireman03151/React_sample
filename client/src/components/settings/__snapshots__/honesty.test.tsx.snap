// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Honesty /> <Honesty /> snapshot when isHonest is false: Honesty 1`] = `
<section
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <p>
        settings.honesty.p1
      </p>
      <p>
        settings.honesty.p2
      </p>
      <p>
        settings.honesty.p3
      </p>
      <p>
        settings.honesty.p4
      </p>
      <p>
        settings.honesty.p5
      </p>
      <p>
        settings.honesty.p6
      </p>
      <p>
        <Trans
          i18nKey="settings.honesty.p7"
        >
          <a
            href="mailto:support@freecodecamp.org"
          >
            {
              "email": "support@freecodecamp.org",
            }
          </a>
        </Trans>
      </p>
    </Uncontrolled(Panel)>
    <Button
      active={false}
      aria-disabled={false}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      buttons.agree-honesty
    </Button>
  </FullWidthRow>
</section>
`;

exports[`<Honesty /> <Honesty /> snapshot when isHonest is true: HonestyAccepted 1`] = `
<section
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <p>
        settings.honesty.p1
      </p>
      <p>
        settings.honesty.p2
      </p>
      <p>
        settings.honesty.p3
      </p>
      <p>
        settings.honesty.p4
      </p>
      <p>
        settings.honesty.p5
      </p>
      <p>
        settings.honesty.p6
      </p>
      <p>
        <Trans
          i18nKey="settings.honesty.p7"
        >
          <a
            href="mailto:support@freecodecamp.org"
          >
            {
              "email": "support@freecodecamp.org",
            }
          </a>
        </Trans>
      </p>
    </Uncontrolled(Panel)>
    <Button
      active={false}
      aria-disabled={true}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      buttons.accepted-honesty
    </Button>
  </FullWidthRow>
</section>
`;
